### PR TITLE
Chronos: Autotsest `future_seq_len` support list input

### PR DESCRIPTION
**Pull request by liangs6212**

### CODE REVIEW

**Comment by liangs6212** [comment]
issue link: #3186

File: python/chronos/src/bigdl/chronos/autots/autotsestimator.py [hunk 1]
```diff
@@ -123,6 +124,11 @@ def __init__(self,
         if isinstance(search_space, str):
             search_space = AutoModelFactory.get_default_search_space(model, search_space)
 
+        self._future_seq_len = future_seq_len  # for support future_seq_len list input.
+        assert not hasattr(future_seq_len, 'sample') and not isinstance(future_seq_len, dict),\
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/autots/autotsestimator.py:128]:
```python
assert isinstance(future_seq_len, (list, int)) # might be wrong
```

File: python/chronos/src/bigdl/chronos/autots/autotsestimator.py [hunk 2]
```diff
@@ -293,7 +299,9 @@ def train_data_creator(config):
             train_d = ray.get(train_data_id)
 
             x, y = train_d.roll(lookback=config.get('past_seq_len'),
-                                horizon=config.get('future_seq_len'),
+                                horizon=self._future_seq_len
+                                if isinstance(self._future_seq_len, list)
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/autots/autotsestimator.py:303]: del line 302-303, 319-319

File: python/chronos/src/bigdl/chronos/autots/autotsestimator.py [hunk 3]
```diff
@@ -123,6 +124,11 @@ def __init__(self,
         if isinstance(search_space, str):
             search_space = AutoModelFactory.get_default_search_space(model, search_space)
 
+        self._future_seq_len = future_seq_len  # for support future_seq_len list input.
+        assert not isinstance(future_seq_len, int) or not isinstance(future_seq_len, list),\
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/autots/autotsestimator.py:128]: del two `not`

File: python/chronos/src/bigdl/chronos/autots/autotsestimator.py [hunk 4]
```diff
@@ -92,8 +92,9 @@ def __init__(self,
         :param past_seq_len: Int or or hp sampling function. The number of historical steps (i.e.
                lookback) used for forecasting. For hp sampling, see bigdl.orca.automl.hp for more
                details. The values defaults to 2.
-        :param future_seq_len: Int. The number of future steps to forecast. The value defaults
-               to 1.
+        :param future_seq_len: Int or List. The number of future steps to forecast. The value
+               defaults to 1, if `future_seq_len` is a list, we will sample discretely according
+               to the input list. 1 means the timestampe just after the observed data.
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/autots/autotsestimator.py:97]: timestamp

File: python/chronos/src/bigdl/chronos/data/tsdataset.py [hunk 5]
```diff
@@ -478,7 +478,7 @@ def roll(self,
                if `horizon` is an int, we will sample `horizon` step
                continuously after the forecasting point.
                if `horizon` is a list, we will sample discretely according
-               to the input list.
+               to the input list. 1 means the timestampe just after the observed data.
```
Comment (TheaperDeng) [inline python/chronos/src/bigdl/chronos/data/tsdataset.py:481]: timestamp

**Review by TheaperDeng — APPROVED** [review]
others LGTM

**Comment by liangs6212** [comment]
http://10.112.231.51:18888/job/BigDL-Chronos-PR-Validation/33/
http://10.112.231.51:18888/job/BigDL-PRVN-chronos-Python-Spark-3.1-py37-ray-part1/52/

**Pull request merged**